mantidproject/mantid
# Allow multiperiod event data to be read by isis sans gui

Fixes #12583 
### For Testing

We should test two things here
1. Do all period workspaces contain the same spectrum numbers (this is also covered in a unit test)
2. Can the multiperiod data be loaded into the SANS GUI and be subsequently reduced.

Note that the relevant files can be found here: \olympic\Babylon5\Scratch\Anton\Testing\12583_Allow_multiperiod_event_data\Sandbox
The user file is: USER_LARMOR_143H_LarmorTeam_80tubes_BenchRot1p5_M1_r3147
The sample data is:LARMOR00002798.nxs
The instrument is LARMOR
Make sure that the path to the sample files is added to your data serach directory.

**Test spectrum numbers**
1. Load the file LARMOR00002798.nxs into Mantid
   - Confirm that a group workspace with four child workspaces is loaded
2. Doble click on each child workspace. 
   - Confirm that all shild workspaces start with the same spectrum id.
   - Confirm that the monitors are not included, ie that the spectrum number starts at 11)

**Test data reduction of multiperiod event data**
1. Open the SANS GUI, set the instrument to LARMOR and load the user file.
2. Close the SANS GUI and reopen it
3. Once everything is loaded, select the data set LARMOR00002798.nxs in the Scattering Sample line edit field. Press "Load Data"
   - Confirm that there are no errors.
4. Press 1D Reduce
   - Confirm that there are no errors

**Test data reduction of multiperiod event data and specification of a single spectrum**
1. Open the SANS GUI and load the user file
2. Check the Multi-period check box and insert 1 next to the Scattering Sample input.
3. Once everything is loaded, select the data set LARMOR00002798.nxs in the Scattering Sample line edit field.  Press "Load Data"
   - Confirm that there are no errors.
4. Press 1D Reduce
   - Confirm that there are no errors

Note that there are currently two types of "new" warnings which appear for this data set. 
- `Cannot find status log. Logs will be not be filtered by run status` which appears because the logs don't contain an entry called "running" and we are trying to load the monitors as well.
- `The base instrument in the output workspace is not the same as the base instrument in the input workspace.` This appears because we seem to have differnt base instruments for the loaded data and the calibartion data set `Vanadium_tub_calib_1to1_30062014`. There might have been a modification on the instrument.

Both warnings seem to be tied rather to the data set than to the new functionality itself.

For release notes see [here](http://www.mantidproject.org/index.php?title=ReleaseNotes_3_5_SANS&diff=24446&oldid=24445)
